### PR TITLE
spelling fixes

### PR DESCRIPTION
**Pull request by rscohn2**
Please look if my spelling fixes are correct for your component

@spencerpatty , @mkrainiuk , @IgorBelyakovIntel @napetrov @outoftardis @krisdale

### CODE REVIEW

File: source/elements/oneDAL/include/oneapi/dal/table/homogen.hpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ class homogen_table : public table {
                   const sycl::vector_class<sycl::event>& dependencies = {},
                   data_layout layout                                  = data_layout::row_major);
 
-    /// Returns the :expr:`data` pointer casted to the $Data$ type. No checks are
+    /// Returns the :expr:`data` pointer castd to the $Data$ type. No checks are
```
Comment (outoftardis) [inline source/elements/oneDAL/include/oneapi/dal/table/homogen.hpp:68]:
```suggestion
    /// Returns the :expr:`data` pointer cast to the $Data$ type. No checks are
```

**Review by outoftardis — COMMENTED** [review]
everything else is ok in onedal component

**Comment by napetrov** [comment]
looks good for oneDAL - assuming Ekaterina's suggestion

**Comment by outoftardis** [comment]
I've also looked at other components, the change in `/onevpl/mfxdefs.h` might require a review, but other changes seem valid

File: source/elements/oneVPL/include/onevpl/mfxdefs.h [hunk 2]
```diff
@@ -187,7 +187,7 @@ MFX_PACK_BEGIN_USUAL_STRUCT()
 typedef struct {
     mfxU32 Min;  /*!< Minimal value of the range. */
     mfxU32 Max;  /*!< Maximal value of the range. */
-    mfxU32 Step; /*!< Value incrementation step. */
+    mfxU32 Step; /*!< Value incrementing step. */
```
Comment (rscohn2) [inline source/elements/oneVPL/include/onevpl/mfxdefs.h:190]: I don't think incrementation is a word. Or if it is, it's not what you mean. incrementing probably isn't right either. How about

increment value

Comment (IgorBelyakovIntel) [reply source/elements/oneVPL/include/onevpl/mfxdefs.h:190]: Agree. That's better.

**Review by mkrainiuk — APPROVED** [review]
Changes for oneMKL look good to me, thank you! I've prepared small fix for spelling in the file name as well https://github.com/rscohn2/oneAPI-spec/pull/1, could you please add it if possible?

**Pull request merged**